src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadSet.cs

```diff
@@ -81,6 +81,45 @@ public static WorkloadSet FromJson(string json, SdkFeatureBand defaultFeatureBan
 #endif
         }
 
+        public static WorkloadSet FromWorkloadSetFolder(string path, string workloadSetVersion, SdkFeatureBand defaultFeatureBand)
+        {
+            WorkloadSet? workloadSet = null;
+            foreach (var jsonFile in Directory.GetFiles(path, "*.workloadset.json"))
+            {
+                var newWorkloadSet = WorkloadSet.FromJson(File.ReadAllText(jsonFile), defaultFeatureBand);
+                if (workloadSet == null)
+                {
+                    workloadSet = newWorkloadSet;
+                }
+                else
+                {
+                    //  If there are multiple workloadset.json files, merge them
+                    foreach (var kvp in newWorkloadSet.ManifestVersions)
+                    {
+                        if (workloadSet.ManifestVersions.ContainsKey(kvp.Key))
+                        {
+                            throw new InvalidOperationException($"Workload set files in {path} defined the same manifest ({kvp.Key}) multiple times");
```

Comment:
This feels like the wrong kind of exception...feels more like a bad argument than an invalid operation? Should it be localized? I know we probably don't care that much, but I could imagine a user seeing it.

Reply:
Ping on localizing this

Reply:
If we throw this message, it means we've authored a workload set incorrectly (or someone has mucked around in dotnet files).  That's the kind of internal error that I don't think we need to localize.